internal/abci/cometbft/peers.go

```diff
@@ -1,10 +1,183 @@
 package cometbft
 
 import (
+	"context"
+	"fmt"
+	"sync"
+
 	"github.com/cometbft/cometbft/crypto/ed25519"
 	"github.com/cometbft/cometbft/p2p"
+	"github.com/kwilteam/kwil-db/common/sql"
+	"github.com/kwilteam/kwil-db/internal/sql/versioning"
+)
+
+const (
+	p2pSchemaName = "peers" // TODO: Umm, should this be part of the chain metadata schema?
+
+	p2pStoreVersion = 0
+
+	initPeersTable = `CREATE TABLE IF NOT EXISTS ` + p2pSchemaName + `.peers (
+		peer_id TEXT PRIMARY KEY
+	);`
+
+	addPeer = `INSERT INTO ` + p2pSchemaName + `.peers ` + `VALUES ($1) ` + `ON CONFLICT (peer_id) DO NOTHING;`
+
+	removePeer = `DELETE FROM ` + p2pSchemaName + `.peers ` + `WHERE peer_id = $1;`
 )
 
+// Peers object is used to manage the set of peers that a node is allowed to connect to.
+// Static peers are those that are added in the genesis and are trusted by default. These
+// include the initial validators, persistent peers, and seed nodes.
+// Whitelist peers are the list of allowed sentry nodes that a node can connect to.
+// Whitelist peers can be updated dynamically using kwil-admin commands and are persisted
+// to disk.
+// The Peers gossiped using PEX are not automatically trusted, and to be added manually if needed.
+// Any new validators are automatically added to the whitelist peers and demoted validators are removed from the whitelist.
+type Peers struct {
```

Comment:
This code all looks great, but maybe rename this to be a `PeerWhitelist` (and change the methods to match that logic)? If I was someone trying to understand what is going on here, I would read this believing that this is the ground truth for what peers are connected, rather than what peers are allowed to connect.